addons/fleet/models/fleet_vehicle.py

```diff
@@ -281,6 +272,16 @@ def action_accept_driver_change(self):
             vehicle.driver_id = vehicle.future_driver_id
             vehicle.future_driver_id = False
 
+    def action_archive(self):
+        self.env['fleet.vehicle.log.contract'].action_archive_with_vehicles(self.ids, True)
+        self.env['fleet.vehicle.log.services'].action_archive_with_vehicles(self.ids, True)
+        super(FleetVehicle, self).action_archive()
+
+    def action_unarchive(self):
+        self.env['fleet.vehicle.log.contract'].action_archive_with_vehicles(self.ids, False)
+        self.env['fleet.vehicle.log.services'].action_archive_with_vehicles(self.ids, False)
+        super(FleetVehicle, self).action_unarchive()
```

Comment:
You can override the ORM method `toggle_active` instead and do what's needed for the contract / services :-)